core/src/main/scala/kafka/security/auth/Authorizer.scala

```diff
@@ -83,6 +83,13 @@ trait Authorizer extends Configurable {
   def getAcls(): Map[Resource, Set[Acl]]
 
   /**
+   * gets set of principal types supported by configured authorizer.
+   * @return set containing only "User" as principal type if Authorizer implementation does not provide this information,
+   *         else set of principal types supported by the configured authorizer.
+   */
+  def getSupportedPrincipalTypes(): Set[String] = Set(KafkaPrincipal.USER_TYPE)
```

Comment:
Since this is just the interface, it probably shouldn't have any implementation.